shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponderTest.mm

```diff
@@ -35,258 +69,216 @@
 }
 }  // namespace
 
-TEST(FlutterChannelKeyResponderUnittests, BasicKeyEvent) {
-  __block NSMutableArray<id>* messages = [[NSMutableArray<id> alloc] init];
-  __block BOOL next_response = TRUE;
-  __block NSMutableArray<NSNumber*>* responses = [[NSMutableArray<NSNumber*> alloc] init];
-
-  id mockKeyEventChannel = OCMStrictClassMock([FlutterBasicMessageChannel class]);
-  OCMStub([mockKeyEventChannel sendMessage:[OCMArg any] reply:[OCMArg any]])
-      .andDo((^(NSInvocation* invocation) {
-        [invocation retainArguments];
-        NSDictionary* message;
-        [invocation getArgument:&message atIndex:2];
-        [messages addObject:message];
-
-        FlutterReply callback;
-        [invocation getArgument:&callback atIndex:3];
-        NSDictionary* keyMessage = @{
-          @"handled" : @(next_response),
-        };
-        callback(keyMessage);
-      }));
+using FlutterChannelKeyResponderTest = AutoreleasePoolTest;
 
+TEST_F(FlutterChannelKeyResponderTest, BasicKeyEvent) {
+  __block NSMutableArray<NSNumber*>* responses = [[NSMutableArray<NSNumber*> alloc] init];
+  FakeMessageChannel* channel = [[FakeMessageChannel alloc] init];
   FlutterChannelKeyResponder* responder =
-      [[FlutterChannelKeyResponder alloc] initWithChannel:mockKeyEventChannel];
+      [[FlutterChannelKeyResponder alloc] initWithChannel:channel];
 
-  // Initial empty modifiers. This can happen when user opens window while modifier key is pressed
-  // and then releases the modifier. No events should be sent, but the callback
-  // should still be called.
+  // Initial empty modifiers.
+  //
+  // This can happen when user opens window while modifier key is pressed and then releases the
+  // modifier. No events should be sent, but the callback should still be called.
   // Regression test for https://github.com/flutter/flutter/issues/87339.
-  [responder handleEvent:keyEvent(NSEventTypeFlagsChanged, 0x100, @"", @"", FALSE, 60)
+  channel.nextResponse = @{@"handled" : @YES};
+  [responder handleEvent:keyEvent(NSEventTypeFlagsChanged, 0x100, @"", @"", NO, 60)
                 callback:^(BOOL handled) {
                   [responses addObject:@(handled)];
                 }];
 
-  EXPECT_EQ([messages count], 0u);
-  EXPECT_EQ([responses count], 1u);
-  EXPECT_EQ([responses[0] boolValue], TRUE);
+  ASSERT_EQ([channel.messages count], 0u);
```

Comment:
(Non-blocking) What's the functional difference here?

Reply:
ASSERT_EQ immediately fails this test case and returns. EXPECT_EQ fails it but continues the rest of this TEST_F block so you'll see any further failures. The issue here is that generally (but not always), when you're checking a count, it's because right after, you'll assume you've got a message and use it, at which point you'd get a crash. 

That's not the case in this line, but it is on the line below.

Reply:
Good catch either way. Updated the test to make this one and the other one checking for a 0 count EXPECT again since they're safe.